### ISSUE TITLE
RSS feed encoding issue

### ISSUE THREAD

**Issue by petira** [getgrav/grav-plugin-feed#70]
RSS feed currently does not correctly transmit characters encoded in UTF-8. Have you made any changes lately?

**Comment by rhukster** [comment]
So to confirm @petira just Atom feed for you? RSS is fine?

**Comment by rhukster** [comment]
So i can recreate the problem. Ie some cyrillic text: `Моя мама любит музыку` in a title, and that is showing garbled in RSS feed and atom feed.  However, I am still not sure why.  It appears to have the correct encoding, and it works fine in the page content itself.  I'm not sure this ever worked?  I went back and tried an older version of Grav, and still didn't work.  Also the feed plugin has not changed in some time.

Are you sure this ever worked guys? or you just noticed it?

**Comment by rhukster** [comment]
I found a solution, needed to set the charset of UTF-8 in the `Content-Type` header.  Unfortunately, there's no way to do that currently in Grav, as the headers are set after the last useful event.  To solve this I had to an event o Grav: `onPageHeaders()` that passes the page headers as an object.  This then allows the feed plugin to use this even to append the `; charset=utf-8`, which then resolves the issue.

This will require this commit in Grav to work: https://github.com/getgrav/grav/commit/de642df06e7e34508bce35f4dd96c7aa34a1d8d2

And this is the commit in feed: https://github.com/getgrav/grav-plugin-feed/commit/bff5a38d24722272cc06b0f7c248786a04c22d59

I think this was not such a huge problem, as while it looks corrupted in the source, if you save it, the encoding is correct, so it probably worked fine in RSS clients.

**Comment by petira** [comment]
@rhukster Thanks for the solution, I will try it once there is a new version.

It is rather strange that it has worked without problems so far. I have been using Feed since the site started running, specifically since December 2020 and sure enough the feeds worked until July 2022, maybe August 2022. It certainly worked even after issue https://github.com/getgrav/grav-plugin-feed/issues/61#issuecomment-849904065 was resolved. I discovered the bug last week.

Here is a link to the feed: https://www.grav.cz/blog.rss

Here is a link to the output: http://www.petira.net/
